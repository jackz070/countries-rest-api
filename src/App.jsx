import React from "react";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import { QueryClient, QueryClientProvider } from "react-query";

import Header from "./components/Header";
import CountriesList from "./components/CountriesList";
import CountryPage from "./pages/CountryPage";

const App = () => {
  const queryClient = new QueryClient();
  return (
    <QueryClientProvider client={queryClient}>
      <BrowserRouter>
        <Header />
        <Routes>
          <Route path="/" element={<CountriesList />} />
          <Route path="/country/:countryName" element={<CountryPage />} />
        </Routes>
      </BrowserRouter>
    </QueryClientProvider>
  );
};

export default App;
