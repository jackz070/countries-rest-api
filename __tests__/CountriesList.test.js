import {
  render,
  screen,
  waitFor,
  act,
  cleanup,
  fireEvent,
} from "@testing-library/react";
import React from "react";
import App from "../src/App";

afterEach(cleanup);

test("renders the homepage", async () => {
  act(() => render(<App />));
  expect(screen.getByText(/Where in the world/i)).toBeInTheDocument();
  expect(screen.getByText(/Mode/i)).toBeInTheDocument();
  expect(screen.getByLabelText(/Loading/i)).toBeInTheDocument();
});

test("dark mode switch works (so far we know that the icon changes on button click, no idea on color", () => {
  act(() => render(<App />));
  expect(document.querySelector("header")).toMatchSnapshot();
  const darkModeSwitch = screen.getByText(/dark mode/i);
  act(() => fireEvent.click(darkModeSwitch));
  expect(document.querySelector("header")).toMatchSnapshot();
});

// TODO try building test env for react-query
