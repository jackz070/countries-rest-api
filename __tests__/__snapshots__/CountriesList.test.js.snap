// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dark mode switch works 1`] = `
<header
  class="flex items-center justify-between sm:h-16 h-24 dark:bg-darkPrimary bg-white shadow-md py-5 sm:px-12 px-4"
>
  <a
    class="font-extrabold text-lg"
    href="/"
  >
    Where in the world?
  </a>
  <div
    class="flex items-center justify-between w-fit h-fit z-[20000]"
  >
    <label
      class=" flex items-center text-xs relative overflow-hidden"
    >
      <input
        class="absolute left-1/2 -translate-x-1/2 w-fit h-fit peer appearance-none rounded-md "
        type="checkbox"
      />
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278zM4.858 1.311A7.269 7.269 0 0 0 1.025 7.71c0 4.02 3.279 7.276 7.319 7.276a7.316 7.316 0 0 0 5.205-2.162c-.337.042-.68.063-1.029.063-4.61 0-8.343-3.714-8.343-8.29 0-1.167.242-2.278.681-3.286z"
        />
      </svg>
      <span
        class="ml-2"
      >
        Dark mode
      </span>
    </label>
  </div>
</header>
`;

exports[`dark mode switch works 2`] = `
<header
  class="flex items-center justify-between sm:h-16 h-24 dark:bg-darkPrimary bg-white shadow-md py-5 sm:px-12 px-4"
>
  <a
    class="font-extrabold text-lg"
    href="/"
  >
    Where in the world?
  </a>
  <div
    class="flex items-center justify-between w-fit h-fit z-[20000]"
  >
    <label
      class=" flex items-center text-xs relative overflow-hidden"
    >
      <input
        class="absolute left-1/2 -translate-x-1/2 w-fit h-fit peer appearance-none rounded-md "
        type="checkbox"
      />
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 .278a.768.768 0 0 1 .08.858 7.208 7.208 0 0 0-.878 3.46c0 4.021 3.278 7.277 7.318 7.277.527 0 1.04-.055 1.533-.16a.787.787 0 0 1 .81.316.733.733 0 0 1-.031.893A8.349 8.349 0 0 1 8.344 16C3.734 16 0 12.286 0 7.71 0 4.266 2.114 1.312 5.124.06A.752.752 0 0 1 6 .278z"
        />
      </svg>
      <span
        class="ml-2"
      >
        Dark mode
      </span>
    </label>
  </div>
</header>
`;
